packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -349,11 +347,23 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       // get fields specified in the schema
       // if there is a role that does not have read access on the field then we create a field resolver
       // or there is a relational directive on the field then we should protect that as well
-      const modelFields = def.fields?.filter(f => acm.hasResource(f.name.value)) ?? [];
+      const readRoles = [...new Set([
+        ...acm.getRolesPerOperation('get'),
+        ...acm.getRolesPerOperation('list'),
+        ...acm.getRolesPerOperation('sync'),
+        ...acm.getRolesPerOperation('search'),
+        ...acm.getRolesPerOperation('listen'),
+      ])];
```

Comment:
Both here and at lines 363-366 we're basically iterating through the list of `read` proxy permissions. It may be more readable and maintainable if we pull those out into a list of type (or hoist your `readExpandedOperations` type below into a constant we can reuse.